tests/mixed_bag.rs

```diff
@@ -527,3 +527,56 @@ fn terse_output() {
             finished in 0.00s
     ");
 }
+
+#[test]
+fn json_output() {
+    let (c, out) = do_run(args(["--format", "json", "--test-threads", "1"]), tests());
+    assert_eq!(
+        c,
+        Conclusion {
+            num_filtered_out: 0,
+            num_passed: 4,
+            num_failed: 4,
+            num_ignored: 8,
+            num_measured: 0,
+        }
+    );
+
+    assert_log!(
+        out,
+        r#"{ "type": "suite", "event": "started", "test_count": 16 }
+{ "type": "test", "event": "started", "name": "cat" }
+{ "type": "test", "name": "cat", "event": "ok" }
+{ "type": "test", "event": "started", "name": "dog" }
+{ "type": "test", "name": "dog", "event": "failed", "stdout": "Error: \"was not a good boy\"\n" }
+{ "type": "test", "event": "started", "name": "fox" }
+{ "type": "test", "name": "fox", "event": "ok" }
+{ "type": "test", "event": "started", "name": "bunny" }
```

Comment:
`assert_log!` cleans indentation, so please indent all these lines (just look at the test above)